### DISCUSSION TITLE
[SOLVED] Need a bit of help building pve-kernel-5.15.83-1

### DISCUSSION THREAD

**Discussion by sarrchri** [Aterfax/relax-intel-rmrr#30] [Q&A] [answered]
Hi!

I made a fork of this and got the kernel to build, but for some reason it built "pve-kernel-6.1.0-1" and not "pve-kernel-5.15.83-1" which I had designated in "build_latest.sh".

Any idea what I did wrong? :D

**Comment by sarrchri** [comment]
Figured it out. Had to build my own docker container, publish it, then build the kernel.

**Comment by Aterfax** [comment, marked as answer]
-----

**In the first instance, I'd recommend that people always run the latest available kernel for security reasons!**

-----


If a specific version of the kernel is desired then you will need to add and execute a new script file which follows the approach in https://github.com/Aterfax/relax-intel-rmrr/blob/master/build/proxmox/build7.1-10.sh 

i.e. :

- Correctly targeting your desired PVE kernel version via `PVE_KERNEL_BRANCH` as well as using the proxmox7.patch. 
- The `RELAX_PATCH` should not be needed since this fork was amended to patch the required iommu source files using `sed` rather than `patch`.

You could then choose to run the build process on your Proxmox machine directly via the CLI or by building a new docker image that you then run where you choose.

If going down the docker image route, you will either need to change or override the entrypoint to run your modified build script rather than the latest available. See the bottom of: https://github.com/Aterfax/relax-intel-rmrr/tree/master/build/proxmox#how-to-do-it-with-docker

I will also highlight where possible, you can avoid having to compile yourselves if the version you need is already available as an older release: https://github.com/Aterfax/relax-intel-rmrr/releases
